File: app/src/main/java/com/test/your_movie/LiveDataHelper.java
package com.test.your_movie;

import android.arch.lifecycle.LiveData;
import android.arch.lifecycle.MediatorLiveData;

import com.test.your_movie.Models.Movie;

import java.util.List;

public class LiveDataHelper {
    private MediatorLiveData<List<Movie>> movieList = new MediatorLiveData<>();
    private static LiveDataHelper liveDataHelper;

    public LiveDataHelper() {}

    synchronized public static LiveDataHelper getInstance() {
        if (liveDataHelper == null)
            liveDataHelper = new LiveDataHelper();
        return liveDataHelper;
    }

    public void updateMovieList(List<Movie> newMoviewList) {
        movieList.postValue(newMoviewList);
    }
    public LiveData<List<Movie>> observeMovieList() {
        return movieList;
    }

}

File: app/src/main/java/com/test/your_movie/Interfaces/LiveDataHelper.kt
package com.test.your_movie.Interfaces

import androidx.lifecycle.LiveData
import androidx.lifecycle.MediatorLiveData

import com.test.your_movie.network.model.movie.Movie

class LiveDataHelper {
    private val movieList = MediatorLiveData<List<Movie>>()

    fun updateMovieList(newMoviewList: List<Movie>) {
        movieList.postValue(newMoviewList)
    }

    fun observeMovieList(): LiveData<List<Movie>> {
        return movieList
    }

//    companion object {
//        private var liveDataHelper: LiveDataHelper? = null
//
//        val instance: LiveDataHelper
//            @Synchronized get() {
//                if (liveDataHelper == null)
//                    liveDataHelper = LiveDataHelper()
//                return liveDataHelper
//            }
//    }

}